File: src/main/java/net/zenxarch/bot/command/ZenCommandManager.java
package net.zenxarch.bot.command;

import net.fabricmc.fabric.api.client.command.v1.ClientCommandManager;

public class ZenCommandManager {
  public static void registerCommands() {
    CraftCommand.register(ClientCommandManager.DISPATCHER);
    PortalCommand.register(ClientCommandManager.DISPATCHER);
  }
}

File: src/main/scala/net/zenxarch/bot/command/ZenCommandManager.scala
package net.zenxarch.bot.command

import net.fabricmc.fabric.api.client.command.v1.ClientCommandManager
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource

object ZenCommandManager:
  val commands = List[ZenCommand](
    new CraftCommand(),
    new PortalCommand(),
    new KillAuraCommand(),
    new MineCommand()
  )
  def registerCommands() =
    commands.foreach(r =>
      ClientCommandManager.DISPATCHER.register(r.register())
    )

abstract class ZenCommand:
  def register(): LiteralArgumentBuilder[FabricClientCommandSource]
